src/tools/rustfmt/src/expr.rs

```diff
@@ -170,8 +170,8 @@ pub(crate) fn format_expr(
                 }
             }
         }
-        ast::ExprKind::Match(ref cond, ref arms) => {
-            rewrite_match(context, cond, arms, shape, expr.span, &expr.attrs)
+        ast::ExprKind::Match(ref cond, ref arms, kind) => {
+            rewrite_match(context, cond, arms, shape, expr.span, &expr.attrs, kind)
```

Comment:
Given that the postfix match is an experimental feature I'd prefer not to add formatting support in this PR. For now I think it makes sene to add just enough support so that rustfmt doesn't completely remove the postfix match.

I think the following changes would achieve that.

```rust
match kind {
    MatchKind::Prefix => rewrite_match(context, cond, arms, shape, expr.span, &expr.attrs),
    MatchKind::Postfix => {
        // experimental feature postfix match (link to RFC)
        Some(context.snippet(expr.span).to_owned())
    }
}
```

cc: @calebcartwright

Reply:
The only way I can think for `context.snippet(expr.span).to_owned()` to crash is if the `expr.span` isn't valid, but that seems unlikely. You could also try returning `None` from the `MatchKind::Postfix` match arm.

Reply:
I actually don't have any concerns with the formatting implementation proposed here as this is rather trivial syntactically, a reasonable initial formatting, and in accordance with the style guide policy on nightly-only/experimental syntax.

the more intriguing part for me was the mention of it "crashing", the details of which I'd be interested in seeing (though that could be shared with us separately in Zulip)

Reply:
Some sort of panic occurred. I'll try to replicate it when I work on this and post it on Zulip. Probably won't be until Friday or Saturday though.

Reply:
Ok, here's a branch with the panic: https://github.com/RossSmyth/rust/tree/rustfmt_test_panic

It may as designed? Looking at it again after just running `x.py test rustfmt`

```ps
Ran 589 system tests.
assertion `left == right` failed: 1 system tests failed
  left: 1
 right: 0
Failed to join a test thread: Any { .. }


failures:
    test::system_tests

test result: FAILED. 171 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 7.15s
```

There are two panics that occur. 

1. https://github.com/rust-lang/rust/blob/279c9ba260389ac92c4827a721d3163c4e519f66/src/tools/rustfmt/src/test/mod.rs#L367-L369

Fails with `left == right failed: 1 idempotent tests failed`

3. https://github.com/rust-lang/rust/blob/279c9ba260389ac92c4827a721d3163c4e519f66/src/tools/rustfmt/src/test/mod.rs#L66-L77

Fails with `panicked at src\\tools\\rustfmt\\src\\test\\mod.rs:76:10:\nFailed to join a test thread: Any { .. }`

Is it designed to just panic if a test fails? It also dumps a pretty large JSON object out.